### PR TITLE
{lib}[GCCcore/13.2.0] libzip v1.10.1

### PR DESCRIPTION
**Pull request by tanmoy1989**
(created using `eb --new-pr`)

### CODE REVIEW

**Review by Micket — APPROVED** [review]
lgtm

**Review by Micket — CHANGES_REQUESTED** [review]
file has execute permissions, it must now, please change this (you may need to do it manually in the branch with git, not sure if EB's pr functionality can do this fix, maybe, maybe not)

**Review by boegel — DISMISSED** [review]

File: easybuild/easyconfigs/l/libzip/libzip-1.10.1-GCCcore-13.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+builddependencies = [
+    ('binutils', '2.40'),
+    ('CMake', '3.27.6'),
+]
+
+dependencies = [('zlib', '1.2.13')]
+
+sanity_check_paths = {
+    'files': [
+        'bin/zipcmp',
```
[the remark this answers is not in the document]
Comment (tanmoy1989) [reply easybuild/easyconfigs/l/libzip/libzip-1.10.1-GCCcore-13.2.0.eb:24]: I am not sure how would I incorporate that? Any suggestions, please?

[the remark this answers is not in the document]
Comment (tanmoy1989) [reply easybuild/easyconfigs/l/libzip/libzip-1.10.1-GCCcore-13.2.0.eb:24]: I have tried now adding sanity check commands. Please let me know if this what you meant or if this correct.

**Comment by tanmoy1989** [comment]
Can we please have update on this?

File: easybuild/easyconfigs/l/libzip/libzip-1.10.1-GCCcore-13.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+easyblock = 'CMakeMake'
+
+name = 'libzip'
+version = '1.10.1'
+
+homepage = 'https://libzip.org/'
+description = "libzip is a C library for reading, creating, and modifying zip archives.."
```
Comment (bedroge) [inline easybuild/easyconfigs/l/libzip/libzip-1.10.1-GCCcore-13.2.0.eb:7]: Not sure what happened in https://github.com/easybuilders/easybuild-easyconfigs/pull/20283/commits/ccb9c6942a2838782d7ef256443ec2260afb7abf, but let's revert this: 
```suggestion
description = "libzip is a C library for reading, creating, and modifying zip archives."
```

**Comment by bedroge** [comment]
@boegelbot please test @ generoso

**Comment by bedroge** [comment]
```
-- Looking for BZ2_bzCompressInit
-- Looking for BZ2_bzCompressInit - found
```
and
```
sanity check command zipmerge -h exited with code 127 (output: zipmerge: error while loading shared libraries: libbz2.so.1.0: cannot open shared object file: No such file or directory
```
seem to suggest that we need to add bzip2 as a dependency.

File: easybuild/easyconfigs/l/libzip/libzip-1.10.1-GCCcore-13.2.0.eb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+source_urls = ['https://github.com/nih-at/libzip/archive/']
+sources = ['v%(version)s.tar.gz']
+checksums = ['d56d857d1c3ad4a7f3a4c01a51c6a6e5530e35ab93503f62276e8ba2b306186a']
+
+builddependencies = [
+    ('binutils', '2.40'),
+    ('CMake', '3.27.6'),
+]
+
+dependencies = [('zlib', '1.2.13')]
```
Comment (bedroge) [inline easybuild/easyconfigs/l/libzip/libzip-1.10.1-GCCcore-13.2.0.eb:20]: So, I suspect this should work:
```suggestion
dependencies = [
    ('zlib', '1.2.13'),
    ('bzip2', '1.0.8'),
    ('XZ', '5.4.4'),
    ('zstd', '1.5.5'),
]
```

**Comment by tanmoy1989** [comment]
Thank you, really appreciate your explanation and suggestion. I accepted the changes you suggested.

**Comment by bedroge** [comment]
@boegelbot please test @ generoso

**Comment by bedroge** [comment]
@boegelbot please test @ jsc-zen3

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bob-Latitude-5300 - Linux Ubuntu 22.04, x86_64, Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz, Python 3.10.12
See https://gist.github.com/bedroge/0ead041f3f227d42eee1e63ff2efa086 for a full test report.

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
interactive1 - Linux Rocky Linux 8.9, x86_64, AMD EPYC-Milan Processor (zen2), Python 3.6.8
See https://gist.github.com/bedroge/eb44015f9825768d82db429748e2d919 for a full test report.

**Comment by bedroge** [comment]
> Test report by @boegelbot **FAILED** Build succeeded for 0 out of 1 (1 easyconfigs in total) jsczen3c1.int.jsc-zen3.fz-juelich.de - Linux Rocky Linux 9.4, x86_64, AMD EPYC-Milan Processor (zen3), Python 3.9.18 See https://gist.github.com/boegelbot/0e979a2277edbe4a1ce0523207a29b2f for a full test report.

Hmm, looks like it also needs OpenSSL. In my build log I also see that it's picking this up.

File: easybuild/easyconfigs/l/libzip/libzip-1.10.1-GCCcore-13.2.0.eb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+sources = ['v%(version)s.tar.gz']
+checksums = ['d56d857d1c3ad4a7f3a4c01a51c6a6e5530e35ab93503f62276e8ba2b306186a']
+
+builddependencies = [
+    ('binutils', '2.40'),
+    ('CMake', '3.27.6'),
+]
+
+dependencies = [
+    ('zlib', '1.2.13'),
```
Comment (bedroge) [inline easybuild/easyconfigs/l/libzip/libzip-1.10.1-GCCcore-13.2.0.eb:21]: Let's try adding OpenSSL 3:
```suggestion
    ('OpenSSL', '3'),
    ('zlib', '1.2.13'),
```

**Comment by bedroge** [comment]
@boegelbot please test @ jsc-zen3

**Comment by bedroge** [comment]
@boegelbot please test @ jsc-zen3

**Comment by bedroge** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/l/libzip/libzip-1.10.1-GCCcore-13.2.0.eb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+sources = ['v%(version)s.tar.gz']
+checksums = ['d56d857d1c3ad4a7f3a4c01a51c6a6e5530e35ab93503f62276e8ba2b306186a']
+
+builddependencies = [
+    ('binutils', '2.40'),
+    ('CMake', '3.27.6'),
+]
+
+dependencies = [
+    ('OpenSSL', '3'),
```
Comment (bedroge) [inline easybuild/easyconfigs/l/libzip/libzip-1.10.1-GCCcore-13.2.0.eb:21]: Oops, forgot to specify the SYSTEM toolchain, and let's stick to 1.1 for now (other easyconfigs in this toolchain are using that version as well):
```suggestion
    ('OpenSSL', '1.1', '', SYSTEM),
```

**Comment by bedroge** [comment]
@boegelbot please test @ jsc-zen3

**Comment by bedroge** [comment]
@boegelbot please test @ jsc-zen3

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bob-Latitude-5300 - Linux Ubuntu 22.04, x86_64, Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz, Python 3.10.12
See https://gist.github.com/bedroge/44605e4352760684671bea43ab30b246 for a full test report.

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
interactive1 - Linux Rocky Linux 8.9, x86_64, AMD EPYC-Milan Processor (zen2), Python 3.6.8
See https://gist.github.com/bedroge/a35a727faff7bd95208e0f029e0f08e2 for a full test report.

**Review by bedroge — APPROVED** [review]
Looks good to me.

**Comment by bedroge** [comment]
Going in, thanks @tanmoy1989!

**Pull request merged**